airflow/models/taskmixin.py

```diff
@@ -161,10 +161,16 @@ def node_id(self) -> str:
 
     @property
     def label(self) -> str | None:
+        from airflow.models.baseoperator import BaseOperator
+        from airflow.models.mappedoperator import MappedOperator
+
         tg = self.task_group
         if tg and tg.node_id and tg.prefix_group_id:
             # "task_group_id.task_id" -> "task_id"
             return self.node_id[len(tg.node_id) + 1 :]
+
+        if isinstance(self, (BaseOperator, MappedOperator)):
+            return self.task_display_name
         return self.node_id
```

Comment:
Hmm... The task group situation is interesting. I think I’d prefer subclassing `label` on BaseOperator and MappedOperator to always use the display name as label instead. In a future release we can add better mechanism on the group level to handle this.

Reply:
Needed to read a lot of code to understand the dependency... and. AGREE here.
Let me have a short regression before committing this as improvement.

Reply:
Additional to this: I believe the current PR lags the feature as well to have a display name on the task groups in general. But would also propose to push this in a follow-up PR. This could contain the cleanup.